src/test/java/studentlist/StudentListTest.java

```diff
@@ -184,4 +203,16 @@ public void findStudent_invalidFindFormat_exceptionThrown() {
 
         assertThrows(SEPException.class, () -> this.studentList.findStudent(input));
     }
+
+    @Test
+    public void filterStudent_nonExistentStudent_exceptionThrown() {
+        // Test filtering an empty student list
+        assertThrows(SEPException.class, () -> this.studentList.findStudent("filter list allocated"));
+    }
+
+    @Test
+    public void filterStudent_invalidFindFormat_exceptionThrown() {
+        String input = "filter list allocate";  // Invalid Filter format ('allocated not allocate')
+        assertThrows(SEPException.class, () -> this.studentList.findStudent(input));
+    }
```

Comment:
great use of unit tests to ensure that the code works as expected! 😄

Reply:
Thank you